sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/PartitionKeyInternal.java

```diff
@@ -259,11 +259,14 @@ public void serialize(PartitionKeyInternal partitionKey, JsonGenerator writer, S
                     return;
                 }
 
-                writer.writeStartArray();
-                for (IPartitionKeyComponent componentValue : partitionKey.getComponents()) {
-                    componentValue.jsonEncode(writer);
+                //  PartitionKey.None has null components - which returns a null list
+                if (partitionKey.getComponents() != null) {
+                    writer.writeStartArray();
+                    for (IPartitionKeyComponent componentValue : partitionKey.getComponents()) {
+                        componentValue.jsonEncode(writer);
+                    }
+                    writer.writeEndArray();
```

Comment:
please add a unit test for the change. See `PartitionKeyInternalTest` for existing tests.

Reply:
Sure, will do.

Reply:
Added unit test.